jenkins/opensearch-dashboards/integ-test.jenkinsfile

```
@@ -144,24 +153,68 @@ pipeline {
                       )
                     sh("cp -a $WORKSPACE/artifacts/${artifactPath} $WORKSPACE")
 
-                    runIntegTestScript(
-                        jobName: "$BUILD_JOB_NAME",
-                        componentName: 'functionalTestDashboards',
-                        buildManifest: "$BUILD_MANIFEST",
-                        testManifest: "manifests/${TEST_MANIFEST}",
-                        localPath: "${WORKSPACE}/${distribution}",
-                        switchUserNonRoot: "${switch_user_non_root}"
-                    )
+                    // Stash the current working directory files, aka opensearch-build repo
+                    // Unstash later in each triggered stage to run integTest
+                    stash includes: "**", name: "integtest-opensearch-dashboards-$BUILD_NUMBER"
+
+                    componentTests = [:]
+
+                    for (component in componentList) {
+                        // Must use local variable due to groovy for loop and closure scope
+                        // Or the stage will be fixed to the last item in return when new stages are triggered here
+                        // https://web.archive.org/web/20181121065904/http://blog.freeside.co/2013/03/29/groovy-gotcha-for-loops-and-closure-scope/
+                        def local_component = component.trim()
+                        def local_component_index = componentList.indexOf(local_component)
+                        def wait_seconds = local_component_index * 10
+
+                        echo "Add Component: ${local_component}"
+                        componentTests["Run Integtest ${local_component}"] = {
+                            // Using scripted pipelines to trigger dynamic parallel stages
+                            timeout(time: 2, unit: 'HOURS') {
+                                node(AGENT_LABEL) {
+                                    docker.withRegistry('https://public.ecr.aws/') {
+                                        docker.image(docker_images["$distribution"]).inside(docker_args["$distribution"]) {
+                                            try {
+                                                stage("Run Integtest ${local_component}") {
```

nit: Everything here is integtest so we can avoid writing the same everywhere
```suggestion
                                                stage("${local_component}") {
```

---

Add the same comment change on OS as well. Thanks.